components/forms/sql/create_shared_schema.sql

```diff
@@ -0,0 +1,101 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+CREATE TABLE IF NOT EXISTS addresses_data (
+    guid          TEXT NOT NULL PRIMARY KEY,
+    given_name          TEXT NOT NULL,
+    additional_name     TEXT NOT NULL,
+    family_name         TEXT NOT NULL,
```

Comment:
@mhammond and I discussed this early on. We decided that we'll use `""` to represent a non-value, but in a hand wavy manner. Asking for my own edification, are you making this suggestion because it's more idiomatic to use `Option` instead of checking a `String`'s value with `is_empty()`? I'm not at all opposed to this change.

Reply:
Yeah, Lougenia and I were waxing lyrical about this. I can't see a practical difference between null and an empty string in these contexts - it's not as though an empty string is a valid `given_name` where NULL would not be, or that we'd treat them differently for conflict resolution purposes, for example - and using non-nullable strings just avoids the extra `Option<>` wrapper which, in all cases I can see, we'd just end up treating as an empty string anyway.

That said though, it was more a philosophical discussion, so I'm open to making them nullable if we can identify a good reason.

Reply:
We decided at yesterday's meeting to keep these `NOT NULL`, and use empty strings to mean empty. Being able to use `String` directly, instead of an `Option<String>`—and then having to worry about what `Some("")` vs. `None` means—was the tipping point.